at_lookup/lib/src/connection/outbound_message_listener.dart

```diff
@@ -27,34 +31,58 @@ class OutboundMessageListener {
   /// Throw a [BufferOverFlowException] if buffer is unable to hold incoming data
   Future<void> messageHandler(data) async {
     String result;
-    if (!_buffer.isOverFlow(data)) {
-      // skip @ prompt
-      if (data.length == 1 && data.first == 64) {
-        return;
-      }
-      // If data does not contain \n and contains @ as last character
-      // then it is partial prompt. Ignore it.
-      // For example if prompt is @alice@, then 'ice@' is received.
-      if ((!data.contains(10)) && data.last == 64) {
-        return;
-      }
-      //ignore prompt(@ or @<atSign>@) after '\n'
-      // If data contains 10(utf8 character for '\n'), trim the string after the '\n'
-      if (data.contains(10)) {
-        data = data.sublist(0, data.lastIndexOf(10) + 1);
-      }
-      _buffer.append(data);
-    } else {
+    // Throw exception if data length exceeds the buffer limit.
+    if (_buffer.isOverFlow(data)) {
+      int bufferLength;
+      (_buffer.length() == 0)
+          ? bufferLength = data.length
+          : bufferLength = (_buffer.length() + data.length) as int;
       _buffer.clear();
       throw BufferOverFlowException(
-          'Buffer overflow on outbound connection result');
+          'data length exceeded the buffer limit. Data length : $bufferLength and Buffer capacity ${_buffer.capacity}');
     }
-    if (_buffer.isEnd()) {
-      result = utf8.decode(_buffer.getData());
-      result = result.trim();
-      _buffer.clear();
-      _queue.add(result);
+    // Add data to buffer until a new line character is encountered
+    _buffer.append(data);
+    // If data does not contains 10, the terminating character is NOT received. so return.
+    if (!data.contains(10)) {
+      return;
+    }
+    result = utf8.decode(_buffer.getData());
+    result = _trimPrompt(result);
+    _queue.add(result);
+    //clear the buffer after adding result to queue
+    _buffer.clear();
+  }
+
+  /// The method accepts the result (server response) and trim's the prompt from the response
+  /// and returns the actual response.
+  String _trimPrompt(String result) {
+    var newLineIndex = result.lastIndexOf('\n');
+    var trimmedResult = result.substring(0, newLineIndex);
+    var prompt = result.substring(newLineIndex).trim();
+    // If isInCompletePromptReceived is true, then partial prompt from previous response
+    // is added to data. trim the prompt from result and reset _isInCompletePromptReceived to false.
+    if (_isInCompletePromptReceived) {
```

Comment:
Should never be possible. The whole point of waiting for newline is that once we receive a full response line terminated by '\n', we *know* that there is a prompt at the start of the line, which we need to trim.

We have two types of info available to us to help us easily trim the prrompt

1) We know that messages from the server, after the prompt portion, always start with `data:` or `error:` or `notification:`. The Java SDK uses this information
2) We know that if we have an unauthenticated connection the prompt will be '@' but if we have an authenticated connection the prompt will be '@<atSign>@'

Reply:
Refactored the code and used 'data:' or 'error:' or 'stream:' to find the start of server responses.